pkg/config/example_test.go

```diff
@@ -0,0 +1,265 @@
+	privKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to generate private key: %v", err))
+	}
+
+	return &config.SigningIdentity{
+		Certificate: generateCert(),
+		PrivateKey:  privKey,
+		MSPID:       "ID",
+	}
+}
+
+// Generate a certificate for the SigningIdentity.
+func generateCert() *x509.Certificate {
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+
+	if err != nil {
+		log.Fatalf("Failed to generate serial number: %s", err)
+	}
+
+	return &x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			CommonName:   "Wile E. Coyote",
+			Organization: []string{"Acme Co"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(365 * 24 * time.Hour),
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+}
+
+func Example_usage() {
+	// Retrieve the config for the channel
+	baseConfig := fetchChannelConfig()
+	// The config that will be updated
+	updatedConfig := proto.Clone(baseConfig).(*cb.Config)
+
+	// Must retrieve the current orderer configuration from block and modify
+	// the desired values
+	orderer := &config.Orderer{
+		OrdererType: config.ConsensusTypeKafka,
+		Kafka: config.Kafka{
+			Brokers: []string{"kafka0:9092", "kafka1:9092"},
+		},
+		Organizations: []*config.Organization{
+			{
+				Name: "OrdererOrg",
+				Policies: map[string]*config.Policy{
+					config.AdminsPolicyKey: {
+						Type: config.ImplicitMetaPolicyType,
+						Rule: "Majority Endorsement",
+					},
+					config.ReadersPolicyKey: { // Adding Readers policy
+						Type: config.ImplicitMetaPolicyType,
+						Rule: "Majority Endorsement",
+					},
+				},
+			},
+		},
+		BatchSize: config.BatchSize{
+			MaxMessageCount:   500, // Updating from 100
+			AbsoluteMaxBytes:  100,
+			PreferredMaxBytes: 100,
+		},
+		Addresses: []string{"127.0.0.1:7050"},
+	}
+
+	err := config.UpdateOrdererConfiguration(updatedConfig, orderer)
+	if err != nil {
+		panic(nil)
+	}
+
+	newAnchorPeer := &config.AnchorPeer{
+		Host: "127.0.0.2",
+		Port: 7051,
+	}
+
+	//	Add a new anchor peer
+	err = config.AddAnchorPeer(updatedConfig, "Org1MSP", newAnchorPeer)
+	if err != nil {
+		panic(err)
+	}
+
+	// Remove an old anchor peer from Org1MSP
+	oldAnchorPeer := &config.AnchorPeer{
+		Host: "127.0.0.2",
+		Port: 7051,
+	}
+
+	err = config.RemoveAnchorPeer(updatedConfig, "Org1MSP", oldAnchorPeer)
+	if err != nil {
+		panic(err)
+	}
+
+	// Compute the delta
+	configUpdate, err := config.ComputeUpdate(baseConfig, updatedConfig, "testChannel")
+	if err != nil {
+		panic(err)
+	}
+
+	// Collect the necessary signatures
+	configSignatures := []*cb.ConfigSignature{}
+
+	peer1SigningIdentity := createSigningIdentity()
+	peer2SigningIdentity := createSigningIdentity()
```

Comment:
Maybe have a comment here indicating the configuration this channel has regarding endorsers/peers etc. The signatures you need to collect here should match the policy required for updating the config properties we chose to update. In other words if you add anchor peers to Org1 as part of this update then you need to satisfy the modification policy for modifying that property (same for every other value we update, ie updating orderer config requires an OrdererOrg admin signature)

Reply:
Eventually when we make modifications to multiple orgs we'll require signatures from a peer from each org.

Reply:
See https://hyperledger-fabric.readthedocs.io/en/release-2.0/config_update.html#get-the-necessary-signatures and https://hyperledger-fabric.readthedocs.io/en/release-2.0/channel_update_tutorial.html#sign-and-submit-the-config-update for examples on which signatures to sign with

Reply:
Added a blurb about how this example is a 2 peer 1 org channel and it meets the defined policies for the configuration.